app/controllers/concerns/two_factor_authenticatable_methods.rb

```diff
@@ -17,7 +17,15 @@ def handle_valid_verification_for_authentication_context(auth_method:)
 
     if IdentityConfig.store.feature_new_device_alert_aggregation_enabled && new_device?
       if current_user.sign_in_new_device_at.blank?
-        current_user.update(sign_in_new_device_at: disavowal_event.created_at)
+        timeframe_expired_at = current_user.events.where(
+          event_type: 'sign_in_notification_timeframe_expired',
+        ).order(created_at: :desc).limit(2)
```

Comment:
Why do we get 2? (`limit(2)`)

Is that brought over the from [my recommendation](https://github.com/18F/identity-idp/pull/10659#discussion_r1620624974) to use `second_last_signed_in_at` as a starting point? I think that doesn't apply here since we're looking for the latest, not the second-to-latest as in `second_last_signed_in_at`.

We might also consider using [`find_by`](https://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find_by) instead of `where`, so that the return value is a single value instead of an array (avoids the `pluck` on the subsequent code)

Reply:
Yeah, but I think you can just tack on `.order` and it should work? I haven't tested.

```rb
current_user.events.order(created_at: :desc).find_by(
  event_type: 'sign_in_notification_timeframe_expired',
)
```

Per the [docs](https://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find_by):

>Finds the first record matching the specified conditions. There is no implied ordering so if order matters, you should specify it yourself.

I think it could be okay with `.limit(1).last` though (or `.limit(1).take`, also [documented as an alternative to `find_by`](https://guides.rubyonrails.org/active_record_querying.html#find-by)).